src/aiida_pseudo/cli/install.py

```diff
@@ -143,117 +140,215 @@ def download_sssp(
-        with open(filepath_archive, 'wb') as handle:
-            handle.write(response.content)
-            handle.flush()
+    for element, values in metadata.items():
+        if family.get_pseudo(element).md5 != values['md5']:
+            Group.collection.delete(family.pk)
+            msg = f"md5 of pseudo for element {element} does not match that of the metadata {values['md5']}"
+            echo.echo_critical(msg)
 
-    with attempt('downloading selected pseudopotentials metadata archive... ', include_traceback=traceback):
-        response = requests.get(url_metadata)
-        response.raise_for_status()
-        with open(filepath_metadata, 'wb') as handle:
-            handle.write(response.content)
-            handle.flush()
+        cutoffs[element] = {'cutoff_wfc': values['cutoff_wfc'], 'cutoff_rho': values['cutoff_rho']}
+
+    family.description = description
+    family.set_cutoffs(cutoffs, 'normal', unit='Ry')
+
+    echo.echo_success(f'installed `{label}` containing {family.count()} pseudopotentials.')
 
 
 @cmd_install.command('sssp')
 @options.VERSION(type=click.Choice(['1.0', '1.1']), default='1.1', show_default=True)
 @options.FUNCTIONAL(type=click.Choice(['PBE', 'PBEsol']), default='PBE', show_default=True)
 @options.PROTOCOL(type=click.Choice(['efficiency', 'precision']), default='efficiency', show_default=True)
 @options.DOWNLOAD_ONLY()
+@options.FROM_DOWNLOAD()
 @options.TRACEBACK()
 @decorators.with_dbenv()
-def cmd_install_sssp(version, functional, protocol, download_only, traceback):
+def cmd_install_sssp(version, functional, protocol, download_only, from_download, traceback):
     """Install an SSSP configuration.
 
     The SSSP configuration will be automatically downloaded from the Materials Cloud Archive entry to create a new
     `SsspFamily`.
     """
-    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-locals, too-many-statements
     from aiida.common.files import md5_file
-    from aiida.orm import Group, QueryBuilder
 
     from aiida_pseudo import __version__
-    from aiida_pseudo.groups.family import SsspFamily
 
-    from .utils import attempt, create_family_from_archive
+    if download_only and from_download is not None:
+        raise click.BadParameter(
+            'cannot specify both `--download-only` and `--from-download`.',
+            param_hint="'--download-only' / '--from-download'"
+        )
 
     configuration = SsspConfiguration(version, functional, protocol)
-    label = SsspFamily.format_configuration_label(configuration)
 
     if configuration not in SsspFamily.valid_configurations:
-        echo.echo_critical(f'{configuration} is not a valid configuration.')
-
-    if not download_only and QueryBuilder().append(SsspFamily, filters={'label': label}).first():
-        echo.echo_critical(f'{SsspFamily.__name__}<{label}> is already installed')
+        echo.echo_critical(f'{version} {functional} {protocol} is not a valid SSSP configuration')
```

Comment:
We should probably move this check to somewhere around after line 221 when the configuration from the archive is also created. Currently that is not validated I believe. Simple validation that the config in the archive is still valid